Signal/src/call/CallAudioService.swift

```diff
@@ -76,89 +111,152 @@ import Foundation
         Logger.verbose("\(TAG) in \(#function) new state: \(call.state)")
 
         switch call.state {
-        case .idle: handleIdle()
-        case .dialing: handleDialing()
-        case .answering: handleAnswering()
-        case .remoteRinging: handleRemoteRinging()
-        case .localRinging: handleLocalRinging()
-        case .connected: handleConnected(call:call)
-        case .localFailure: handleLocalFailure()
-        case .localHangup: handleLocalHangup()
-        case .remoteHangup: handleRemoteHangup()
-        case .remoteBusy: handleBusy()
+        case .idle: handleIdle(call: call)
+        case .dialing: handleDialing(call: call)
+        case .answering: handleAnswering(call: call)
+        case .remoteRinging: handleRemoteRinging(call: call)
+        case .localRinging: handleLocalRinging(call: call)
+        case .connected: handleConnected(call: call)
+        case .localFailure: handleLocalFailure(call: call)
+        case .localHangup: handleLocalHangup(call: call)
+        case .remoteHangup: handleRemoteHangup(call: call)
+        case .remoteBusy: handleBusy(call: call)
         }
     }
 
-    private func handleIdle() {
+    private func handleIdle(call: SignalCall) {
         Logger.debug("\(TAG) \(#function)")
     }
 
-    private func handleDialing() {
+    private func handleDialing(call: SignalCall) {
         Logger.debug("\(TAG) \(#function)")
+
+        if call.isSpeakerphoneEnabled {
+            setAudioSession(category: AVAudioSessionCategoryPlayAndRecord,
+                            mode: AVAudioSessionModeVoiceChat,
+                            options: [.defaultToSpeaker, .mixWithOthers])
```

Comment:
Not enums exactly, but options yes: https://developer.apple.com/reference/swift/optionset